drivers/hmis/app/models/hmis/form/form_processor.rb

```diff
@@ -401,6 +426,25 @@ def related_records
     end.compact.uniq
   end
 
+  def destroy_related_records!
```

Comment:
Might want to add a comment here that this will need to be updated along with the data standards.  I'm tempted to suggest we move this list into `HudUtility2024` to enforce that.

Reply:
Not sure about that one, I don't see why we should limit ourselves to say that only HUD records can be updated from forms. You might want to have an assessment that generates a Referral for example, or some other kind of custom record.

Reply:
I wasn't proposing that they'd have to be HUD tables, only that we _know_ the HUD ones change on a schedule, so maybe we should move those there.  something like:
```
tables = [
  :referrals,
] + HudUtility2024.enrollment_dependent_hmis_tables
tables.map { |assoc| send(assoc)&.destroy! }.compact
```
maybe?